hudi-common/src/main/avro/HoodieIndexPartitionInfo.avsc

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+{
+  "namespace": "org.apache.hudi.avro.model",
+  "type": "record",
+  "name": "HoodieIndexPartitionInfo",
+  "fields": [
+    {
+      "name": "version",
+      "type": [
+        "int",
+        "null"
+      ],
+      "default": 1
+    },
+    {
+      "name": "metadataPartitionPath",
+      "type": [
+        "null",
+        "string"
+      ],
+      "default": null
+    },
+    {
+      "name": "dataPartitionPath",
```

Comment:
Where is this data partition path going to be used? Will all index partition have metadata + data partition combo?

Reply:
We don't really need this. We can find the concerned data partitions from the last completed instant. I added this with the following use-case in mind: let's say users have jobs running once a day/hour and it's append-only data partitioned by day/hour. In that case, users can specify upto what data partitions they want metadata index to be created. Maybe they don't want all data partitioned to be indexed to save time.

But, storing data partitions in commit metadata will incur serde costs which we cannot ignore because index metadata is going to be read for every write. Moreover, metadata is not exposed to the user. I will remove it.